pkg/api/persistentvolume/util.go

```diff
@@ -28,6 +28,21 @@ func DropDisabledFields(pvSpec *api.PersistentVolumeSpec, oldPVSpec *api.Persist
 	if !utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) && !volumeModeInUse(oldPVSpec) {
 		pvSpec.VolumeMode = nil
 	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ExpandCSIVolumes) && !hasExpansionSecrets(oldPVSpec) {
+		pvSpec.CSI = nil
```

Comment:
should this only nil out the ExpandSecretRef instead of the entire csi spec?

Reply:
*head desk*